### ISSUE TITLE
When rewriting paths, remove lineto segments that are redundant with a `z` command

### ISSUE THREAD

**Issue by tigt** [svg/svgo#1038]
Here's a reduced test case:

```xml
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256">
  <path d="M20 20l80 80H20V20z"/>
</svg>
```

The penultimate `V20` command draws to the same point as the origin, which is what `z` does anyway. Removing `V20` should be safe.

There is a similar case when a redundant Lineto command directly follows a Moveto, like this:

```xml
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 256 256">
  <path d="M20 100V20a1 1 0 0 1 0 80z"/>
</svg>
```

In this case, if the initial Moveto is changed to `M20 20` and the following `V20` is removed, the rendering is identical.

**Comment by tigt** [comment]
Er, when I said "origin" in the original post, I meant the originating `M` command at the beginning — not 0,0

**Comment by GreLI** [comment]
IIRC there are some cases when it’s needed, like some detail on stroke rendering. But yeah that’s a possible optimization.

**Comment by tigt** [comment]
Fair point. I'll make some test cases for shapes with strokes and see if I can identify what changes
